nomos-services/mempool/src/backend/mockpool/mod.rs

```diff
@@ -31,20 +82,37 @@ where
     }
 }
 
+impl<BlockId, Item, Key> Clone for MockPool<BlockId, Item, Key>
+where
+    Key: Hash + Eq + Clone,
+    Item: Clone,
+    BlockId: Clone,
+{
+    fn clone(&self) -> Self {
+        Self {
+            pending_items: self.pending_items.clone(),
+            in_block_items: self.in_block_items.clone(),
+            in_block_items_by_id: self.in_block_items_by_id.clone(),
+            last_item_timestamp: self.last_item_timestamp,
+        }
+    }
+}
+
 impl<BlockId, Item, Key> MockPool<BlockId, Item, Key>
 where
     Key: Hash + Eq + Clone,
 {
+    #[must_use]
     pub fn new() -> Self {
-        Default::default()
+        Self::default()
```

Comment:
Is there a lint for this? I've seen this around in other places like phantoms and I'm curious why the change?

Reply:
Yes. `'-W clippy::default-trait-access' implied by '-W clippy::pedantic'`

Reply:
I wonder at what point did this changes. Using `Default` was the pattern at some point iirc. Anyway, doesn't really matter too much. Thanks!

Reply:
I think `..Default::default()` is expected when we want to initialise some fields and use the type's `Default` implementation for the rest. If we need the whole type's default, from within the same type impl block, then this has been there for a while, as far as I remember.